metricflow/model/semantics/semantic_model_lookup.py

```diff
@@ -141,6 +141,19 @@ def get_measure(self, measure_reference: MeasureReference) -> Measure:  # noqa:
             semantic_model=semantic_model, measure_reference=measure_reference
         )
 
+    def get_measure_with_agg_time_dimension(self, measure_reference: MeasureReference) -> Measure:  # noqa: D
```

Comment:
Sorry, I'm not clear on the context behind this change, so this may be a silly question, but do we need this method? For other cases where we need to do some kind of custom stuff at the engine API level we have custom objects. It might be simplest to add a Measure type and then populate the agg_time_dimension there, similar to what we do with Metric: https://github.com/dbt-labs/metricflow/blob/main/metricflow/engine/models.py#L27-L58

Reply:
Yeah this is a question I had and I think my assumption was just wrong - I assumed that it must be intentional that we were not populating `agg_time_dimension` on `Measure`. If that's not the case, I wonder if we should be populating `agg_time_dimension` whenever we build a `Measure` object? It seems weird to have an `agg_time_dimension` field available on `Measure` anywhere in the code if it is not always properly populated.

Reply:
Also I'm not sure if it was clear to you why I did this, so I'll explain:
The `Measure` object has an `agg_time_dimension` field, always. This is not a field I'm adding in MFEngine. But that field is not always populated. In the case when the `agg_time_dimension` is inherited from the default specified for the semantic model, that field returns `null`. Is that intentional? If not, my thought is that we should do what I mentioned above and make sure it's populated anytime we build a `Measure` object.

Reply:
It's not practical to always populate this parameter because of the way dbt core does these model default -> sub-object transformations, so we pushed the "always populate" logic into the outside accessor you're using. This makes everything difficult, and untangling the whole mess is fairly involved.

What you really need is a version of Measure with agg_time_dimension set to required instead of optional. I think the simplest way to get it is to do what we're doing with Metric and Dimension and make an engine-level dataclass representing the Measure properties for now. You can make agg_time_dimension required on that object, and then populate it from here without needing any API expansion on the SemanticModelLookup class. That way you don't ever need to deal with the Optional nature of the property on the MetricFlow Server side, and we have at least some typechecker support around our contract for the properties we include on these objects.

Perhaps we'll consolidate things later, although I suspect we'll live like this forever. :/

Reply:
> What you really need is a version of Measure with agg_time_dimension set to required instead of optional.

Yeah that makes sense to me! Cool I can do it that way.